XRTK.SDK/Packages/com.xrtk.sdk/Runtime/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -184,7 +188,14 @@ protected virtual void OnTriggerEnter(Collider other)
                 nearInteractionCollider != null &&
                 other.IsValidCollision(PointerRaycastLayerMasksOverride ?? MixedRealityToolkit.InputSystem.FocusProvider.GlobalPointerRaycastLayerMasks))
             {
-                capturedNearInteractionObject = other.gameObject;
+                CapturedNearInteractionObject = other.gameObject;
+
+                // Force update the focus provider so the focused target
+                // gets updated before raising the event. If we don't update
+                // the focus provider here, the event will not be raised on the
+                // capture near interaction object.
+                MixedRealityToolkit.InputSystem.FocusProvider.Update();
```

Comment:
@StephenHodgson when you have some time could you please comment on this. I don't like this change at all but it's the only way to currently make near interaction work.

I didn't see a way to make it work without touching the FocusProviders itself and other components of the input system. In general I think we need to stop making these ad-hoc changes for near interaction and come up with a proper concept and data flow for near interaction to work. It doesn't feel quite there yet. I think what you did so far is a good starting point but upon debugging I found there is still more work to do to make this work. It's not enough to just register `OnTriggerEnter()` etc. We need to properly handle the focus change as well.

Reply:
Essentially what happens here is that I have to force an update / raycast on the focus provider before `RaiseOnInput...`. If I don't do this, the input event will not be raised on `CapturedNearInteractionObject` but instead on whatever the pointer was focusing before.